### PR TITLE
Views

### PR DESCRIPTION
**Pull request by bkalika**

### CODE REVIEW

**Comment by bkalika** [comment]
[Homework #14. Django Views. KALIKA BOHDAN](https://trello.com/c/vTFyiQKl/294-homework-14-django-views-kalika-bohdan)

File: app/concierge/mycore/templates/tenant.html [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{% extends "index.html" %}
+
+{% block form %}
+<h2>Tenant</h2>
+<form action="/tenant/" method="POST" enctype="multipart/form-data">
```
Comment (dimigor) [inline app/concierge/mycore/templates/tenant.html]: use `url route_name` like above

File: app/concierge/mycore/tests.py [hunk 2]
```diff
@@ -1,16 +1,128 @@
...
+# class KeyModelTests(TestCase):
+#     def test_key_model_creation_positive_case(self):
+#         room = Room(number=2)
+#         room.save()
+#         key = Key(room=room)
+#         key.save()
+#
+#         keys = Key.objects.all()
+#         self.assertEqual(key.room.number, 2)
+#         self.assertEqual(keys[0].id, 4)
```
Comment (dimigor) [inline app/concierge/mycore/tests.py]: :'(

File: app/concierge/mycore/views.py [hunk 3]
```diff
@@ -1,3 +1,211 @@
...
+        data = {"rooms": rooms, 'form': form, 'owner': owner}
+        return render(request, self.template_name, data)
+
+    def post(self, request):
+        form = self.form_class(request.POST)
+        file = self.form_class(request.FILES)
+        context = {'form': form}
+        if form.is_valid():
+            form.save_form()
+            return HttpResponseRedirect('/room', context)
```
Comment (dimigor) [inline app/concierge/mycore/views.py]: use `reverse`

File: app/concierge/concierge/settings.py [hunk 4]
```diff
@@ -28,8 +27,8 @@
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = True
 
-ALLOWED_HOSTS = ["0.0.0.0"]
-
+# ALLOWED_HOSTS = ["0.0.0.0"]
+ALLOWED_HOSTS = ["127.0.0.1", "0.0.0.0:8000", "0.0.0.0", "localhost"]
```
[the remark this answers is not in the document]
Comment (bkalika) [reply app/concierge/concierge/settings.py]: ok

File: app/concierge/concierge/templates/base.html [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Base</title>
+</head>
+<body>
+<h1>Base html</h1>
+
+</body>
+</html>
```
[the remark this answers is not in the document]
Comment (bkalika) [reply app/concierge/concierge/templates/base.html]: I have forgotten. Already have done.

File: app/concierge/mycore/models.py [hunk 6]
```diff
@@ -13,7 +13,15 @@ class Tenant(Model):
...
-    photo = ImageField('Photo', upload_to='tenant', help_text='Photo of the tenant', null=True, blank=True)
+    photo = ImageField('Photo',
+                       upload_to='tenant',
+                       help_text='Photo of the tenant',
+                       null=True,
+                       blank=True,)
+                       # width_field="width_field",
+                       # height_field="height_field")
+    # height_field = IntegerField(default=3456)
+    # width_field = IntegerField(default=5184)
```
[the remark this answers is not in the document]
Comment (bkalika) [reply app/concierge/mycore/models.py]: I left this is for me (in advance like cheat-sheet).

File: app/concierge/mycore/templates/tenant.html (hunk 1 above)
Comment (bkalika) [reply app/concierge/mycore/templates/tenant.html]: I do not understand you.

File: app/concierge/mycore/tests.py (hunk 2 above)
Comment (bkalika) [reply app/concierge/mycore/tests.py]: this I will do in next homework.

File: app/concierge/mycore/views.py (hunk 3 above)
Comment (bkalika) [reply app/concierge/mycore/views.py]: have done.

File: app/concierge/mycore/migrations/0001_initial.py [hunk 7]
```diff
@@ -1,6 +1,7 @@
-# Generated by Django 3.0.1 on 2019-12-23 18:05
+# Generated by Django 3.0.1 on 2020-01-26 17:34
```
Comment (dimigor) [inline app/concierge/mycore/migrations/0001_initial.py]: never update created migration. Always create new one

**Pull request merged**